MDANSE_GUI/Src/MDANSE_GUI/Widgets/PlotSettingsDialog.py

```diff
@@ -0,0 +1,339 @@
+#    This file is part of MDANSE_GUI.
+#
+#    MDANSE_GUI is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+import traceback
+from typing import Any
+
+from matplotlib import rc, rc_file, rcdefaults, rcParams, rcParamsDefault
+from qtpy.QtCore import QObject, QSortFilterProxyModel, Qt, Signal, Slot
+from qtpy.QtGui import QColor, QStandardItem, QStandardItemModel
+from qtpy.QtWidgets import (
+    QApplication,
+    QComboBox,
+    QDialog,
+    QHBoxLayout,
+    QLabel,
+    QLineEdit,
+    QPushButton,
+    QTreeView,
+    QVBoxLayout,
+)
+
+from MDANSE import PLATFORM
+from MDANSE.MLogging import LOG
+
+ERROR_COLOR = QColor(255, 0, 0)
+
+
+def parse_string(param_str: str) -> Any:
+    """Convert a matplotlib rcParams value to a Python object.
+
+    Parameters
+    ----------
+    param_str : str
+        String obtained from rcParams.values()
+
+    Returns
+    -------
+    Any
+        Typically a string, list[float], list[str] or None
+    """
+    param_str = str(param_str).replace("\\", "")
+    param_str = param_str.replace("'", "")
+    if "[" in param_str:
```

Comment:
The functionality below only makes sense if:
```suggestion
    if param_str.startswith("["):
```